sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -256,15 +259,15 @@ public <T> T insert(String containerName, T objectToSave, PartitionKey partition
     @SuppressWarnings("unchecked")
     public <S extends T, T> Iterable<S> insertAll(CosmosEntityInformation<T, ?> information, Iterable<S> entities) {
         Assert.notNull(entities, "entities to be inserted should not be null");
-
         String containerName = information.getContainerName();
         Class<T> domainType = information.getJavaType();
-
         List<CosmosItemOperation> cosmosItemOperations = new ArrayList<>();
+        Map<String, Object> entitiesMap = new HashMap<>();
         entities.forEach(entity -> {
             markAuditedIfConfigured(entity);
             generateIdIfNullAndAutoGenerationEnabled(entity, domainType);
-            JsonNode originalItem = mappingCosmosConverter.writeJsonNode(entity);
+            JsonNode originalItem = mappingCosmosConverter.writeJsonNode(entity, true);
+            entitiesMap.put(originalItem.get("id").asText(), entity);
```

Comment:
At the time, I considered this to be moot, because items will fail to even write to the database if they don't have an id, and I doubt anyone would want id to be transient. But, really we should explicitly throw a cosmos exception with message if either the id or partitionKey field has been marked `@Transient` - that should be dissallowed. I'll add that.

Reply:
Oh I see you are asking a different question - and the answer to your question is yes, the json that is inserted to Cosmos DB will still have to have an id field, and it will have to be present in JsonNode. But this does make me realise that field marked with both `@Transient` and @PartitionKey` cannot have that field removed from the JsonNode. Also, if the field "id" is marked as either `@Id` or `@PartitionKey`, it also should not be removed. Need to add logic and tests for that.

Reply:
I've added logic and tests but thought it better to add warnings rather than throw exceptions if attempting to add `@Transient` to `id` or `PartitionKey`.